packages/components/src/base-control/index.js

```diff
@@ -16,4 +16,14 @@ function BaseControl( { id, label, help, className, children } ) {
 	);
 }
 
+BaseControl.VisualLabel = ( { className, children } ) => {
+	return (
+		<div className={
+			classnames( 'components-base-control-visual-label', className )
+		}>
```

Comment:
Minor: We don't have much guideline around JSX formatting, but in my experience I tend to see mostly one of either:

```jsx
<div className="its-a-single-line">
```

Or:

```jsx
<div
	className="its-multiple-lines"
>
```

Where if in the multi-line, the value of the attribute spanned multiple lines, it would look more like:

```jsx
<div
	className={
		'its-even-more-lines'
	}
>
```

In this case, I think all of it could be avoided by just assigning the result of `classnames` into a variable, depending on our level of comfort with overriding the value of the `className` prop:

```jsx
className = classnames( 'components-base-control-visual-label', className );

return (
	<div className={ className }>
```

I don't feel very strongly about any of this, though I've thought on a few occasions whether it's worth starting a discussion on more rigid conventions around JSX formatting.

Reply:
Hi @aduth,
I agree maybe we can analyse a stricter JSX formatting rule.
Although we normally prefer this formats:
```
<div className="its-a-single-line">
```
```
<div
	className="its-multiple-lines"
>
````

The other format has an advantage regarding internacionalization -- I think it is the one that guarantees translator comments correctly reference the string.
<MyComponent
	mySpecialLabel={
		// the translator comment
		__( 'my special label' )
	}
>

I noticed it is even safer than using variables before e.g:
// the translator comment
const mySpecialLabel = __( 'my special label' )

Because during the transpiling the variable declarations may be reordered/set in a different way.


But in this case, the usage of a variable is the one that makes more sense. The code was updated.